examples/http/onvif_camera.js

```diff
@@ -0,0 +1,224 @@
+/**
+ * Domotz Custom Driver 
+ * Name: ONVIF camera monitoring
+ * Description: Monitors the video capturing status of an ONVIF camera to check if capturing is still working or has stopped.
+ * 
+ * Communication protocols is Telnet and HTTP
+ * 
+ * Tested on LILIN camera model LD2222
+ *
+ * Creates a Custom Driver table  with video capturing status for each camera profiles
+ * 
+ **/
+
+// Device credentials
+var username = D.device.username();
+var password = D.device.password();
+
+// Encoding credentials for authorization
+var authString = username + ":" + password;
+var base64AuthString = D._unsafe.buffer.from(authString).toString('base64');
+
+// HTTP request parameters
+var url = "/onvif/device_service";
+
+var filterProfileName = D.getParameter("profileName");
+
+// Telnet parameters
+var telnetParams = {
+    negotiationMandatory: false,
+    port: 554
+};
+
+// Table for capturing status
+var table = D.createTable(
+    "Video Streaming Status",
+    [
+        { label: "Streaming status" }
+    ]
+);
+
+/**
+ * Retrieves the profiles from the ONVIF camera.
+ * @returns A promise that resolves with an array of profile tokens.
+ */
+function getProfiles() {
+    var d = D.q.defer();
+    var config = {
+        url: url,
+        username: username,
+        password: password,
+        auth: "basic",
+        body: '<?xml version="1.0" encoding="UTF-8"?>\n<soap:Envelope xmlns:soap="http://www.w3.org/2003/05/soap-envelope" xmlns:trt="http://www.onvif.org/ver10/media/wsdl">\n<soap:Body>\n<trt:GetProfiles/>\n</soap:Body>\n</soap:Envelope>' 
```

Comment:
you are missing the security header


    <s:Header><Security s:mustUnderstand="1" xmlns="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-secext-1.0.xsd"><UsernameToken><Username>admin</Username><Password Type="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-username-token-profile-1.0#PasswordDigest">W/1f5yAYWflLrwUSWT/GVrZnGAk=</Password><Nonce EncodingType="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-soap-message-security-1.0#Base64Binary">7qqG1t66wYq8/E2SStjqLw==</Nonce><Created xmlns="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-utility-1.0.xsd">2024-05-16T12:36:17.000Z</Created></UsernameToken></Security>
    </s:Header>